### PR TITLE
FF-A map manifest region

### PR DESCRIPTION
**Pull request by jellesels-arm**
Map the device and memory regions from the SP manifest file into the SP memory space.
Iterate over all device and manifest regions in the manifest file. Map the regions inside the SP memory space.
With this patch the Trusted Service crypto can use the hardware TRNG.

### CODE REVIEW

File: core/arch/arm/kernel/secure_partition.c [hunk 1]
```diff
@@ -299,7 +301,28 @@ TEE_Result sp_map_shared(struct sp_session *s,
...
+
 	SLIST_FOREACH(reg, &smem->regions, link) {
+		/* Check device region attributes */
+		if (is_device) {
+			uint32_t mcache_type = 0;
+
+			if (mobj_get_cattr(reg->mobj, &mcache_type))
+				return FFA_INVALID_PARAMETERS;
+
+			if (cache_type != mcache_type) {
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/secure_partition.c:377]: Couldn't this check have been performed earlier when the smem was created? It seems that the smem is in an inconsistent state that never should have been allowed if this test fails.

**Review by jenswi-linaro — COMMENTED** [review]
Comment for the commit "core: Allow SPs to map share device memory"

File: core/arch/arm/mm/sp_mem.c [hunk 2]
```diff
@@ -126,12 +125,25 @@ void  sp_mem_set_cattr(struct mobj *mobj, uint32_t cache_type)
...
+
+	m->is_secure = is_secure;
+}
+
 static bool mobj_sp_matches(struct mobj *mobj __maybe_unused,
 			    enum buf_is_attr attr)
 {
+	struct mobj_sp *m = to_mobj_sp(mobj);
+
 	assert(mobj->ops == &mobj_sp_ops);
```
Comment (jenswi-linaro) [inline core/arch/arm/mm/sp_mem.c:158]: This assert can be removed, it's done inside `to_mobj_sp()`.

File: core/arch/arm/mm/sp_mem.c [hunk 3]
```diff
@@ -82,9 +84,6 @@ int sp_mem_add_pages(struct mobj *mobj, unsigned int *idx,
 	if (ADD_OVERFLOW(*idx, num_pages, &n) || n > tot_page_count)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	if (!core_pbuf_is(CORE_MEM_NON_SEC, pa, num_pages * SMALL_PAGE_SIZE))
```
Comment (jenswi-linaro) [inline core/arch/arm/mm/sp_mem.c:82]: We will need some way of checking that the security state of the `mobj_sp` matches the one of the physical pages represented.

**Review by jenswi-linaro — COMMENTED** [review]
Comments for "core: sp_mem: Allow secure memory mapping"

File: core/arch/arm/mm/sp_mem.c (hunk 2 above)
Comment (jellesels-arm) [reply core/arch/arm/mm/sp_mem.c]: Ack

File: core/arch/arm/mm/sp_mem.c (hunk 3 above)
Comment (jellesels-arm) [reply core/arch/arm/mm/sp_mem.c:82]: Ack

File: core/arch/arm/kernel/secure_partition.c (hunk 1 above)
Comment (jellesels-arm) [reply core/arch/arm/kernel/secure_partition.c]: This check is not needed like you are saying, smem is the one which is setting the cache_type before.

**Comment by jforissier** [comment]
Superseded by https://github.com/OP-TEE/optee_os/pull/5223.

**Pull request closed without merging**